client/src/components/dashboard.js

```diff
@@ -41,6 +57,19 @@ function Dashboard() {
                   <td className="channel__count">
                     {channel.members.length + 1}
                   </td>
+                  <td>
+                    {!isChannelMember(channel) && !isChannelOwner(channel) ? (
+                      <button
+                        className="btn btn-info"
+                        id={channel._id}
+                        onClick={joinChannel}
```

Comment:
instead of using `e.target.id` while adding this line, we can just use `channel._id` directly in `joinChannel`